modules/ml/test/test_rtrees.cpp

```diff
@@ -95,6 +95,24 @@ TEST(ML_RTrees, 11142_sample_weights_classification)
     EXPECT_GE(error_with_weights, error_without_weights);
 }
 
+TEST(ML_RTrees, bug_12974_throw_exception_when_predict_different_feature_count)
+{
+    // create a simple 2 feature dataset and train + save the model
+    cv::Ptr<RTrees> model = RTrees::create();
+    Mat samples(2, 2, CV_32FC1);
+    samples.at<float>(0,0) = 0;
```

Comment:
Matrices can be initialized in more compact way, check out the previous test.

Reply:
Good call - funny enough, I was also the one who wrote the previous test :)

Reply:
Addressed